File: code/assignment/week03/no4_java/Warrior.java
public class Warrior {
    // 武士类型
    private String type;
    // 武士的初始生命值，其实就是「力量」
    private int strength;

    public Warrior(String type, int strength) {
        this.type = type;
        this.strength = strength;
    }

    public String getType() {
        return type;
    }

    public int getStrength() {
        return strength;
    }
}

File: code/assignment/week03/no4/Warrior.cpp
#include <string>

using namespace std;

class Warrior {
  // 武士类型
  string type;
  // 武士的初始生命值，其实就是「力量」
  int strength;

public:
  Warrior(string type, int strength);
  string getType();
  int getStrength();
};

Warrior::Warrior(string type, int strength) {
  this->type = type;
  this->strength = strength;
}

string Warrior::getType() { return this->type; }

int Warrior::getStrength() { return this->strength; }